src/algebra/algebra.jl

```diff
@@ -1,5 +1,34 @@
 otimes(x::AbstractVector, y::AbstractVector) = x * y'
 otimes(x::AbstractVector) = otimes(x, x)
+
+"""
+Tensors product between second order tensors
+
+# Implementation
+A[i,j] ⊗ B[k,l] = C[i,j,k,l]
+such as
+C[i,j,k,l] = A[i,j] * B[k,l]
+"""
+function otimes(A::AbstractArray{T1, 2}, B::AbstractArray{T2, 2}) where {T1, T2}
+    sA = size(A)
+    sB = size(B)
+    C = zeros(sA[1], sA[2], sB[1], sB[2])
+    for k in 1:sB[1]
+        for l in 1:sB[2]
+            C[:, :, k, l] = A .* B[k, l]
+        end
+    end
+    return C
+end
+
+# otimes for static arrays
+function otimes(
+    A::SMatrix{I1, I2, T1, L1},
+    B::SMatrix{I3, I4, T2, L2},
+) where {I1, I2, I3, I4, T1, T2, L1, L2}
+    return SArray{Tuple{I1,I2,I3,I4}}([A[i, j] * B[k, l] for i in 1:I1, j in 1:I2, k in 1:I3, l in 1:I4])
```

Comment:
I used the same formalism as the dcontract functions.
I don't understand why it would be necessary to add a line to retrieve the sizes when they are retrieved directly from the input matrices, which are static matrices.
For exemple :
```julia
function dcontract(
    A::SArray{<:Tuple{I1, J, K}, T1, 3, L1},
    B::SArray{<:Tuple{I2, J, K}, T2, 3, L2},
) where {I1, J, K, I2, T1, T2, L1, L2}
    return SMatrix{I1, I2}(sum(A[i, :, :] .* B[j, :, :]) for i in 1:I1, j in 1:I2)
end
```

Reply:
I was talking about the case where arrays are not static. In other words, you can implement the non-static case with a list-comprehension, as you did for the static case